src/main/java/org/xwiki/git/GitManager.java

```diff
@@ -48,6 +48,24 @@
      */
     Repository getRepository(String repositoryURI, String localDirectoryName);
 
+    /**
+     * Clone a Private Git repository using the credentials provided by user and store it locally in the
+     * XWiki Permanent directory. If the repository is already cloned, no action is done.
+     *
+     * @param repositoryURI the URI to the Git repository to clone (eg "git://github.com/xwiki/xwiki-commons.git")
+     * @param localDirectoryName the name of the directory where the Git repository will be cloned (this directory is
+     *        relative to the permanent directory
+     * @param username the username of Git user
+     * @param token the Personal access token generated by the user
```

Comment:
Maybe mention how to get that token. Hmmm what token is this about?

Reply:
Updated.